### PR TITLE
refactor api, use localstorage if available

### PR DESCRIPTION
**Pull request by prestonp**
On my macbook air, it shaves the front page from ~5 secs to ~3 secs.

### CODE REVIEW

**Comment by luqmaan** [comment]
Thanks pham

File: client/js/utils/api.js [hunk 1]
```diff
@@ -1,72 +1,23 @@
...
-
-        if (_cache[url]) {
-            return when.resolve(_cache[url]);
-        }
-
-        return ajax({url: url, dataType: 'json'})
-            .tap((body) => _cache[url] = body)
-            .catch((err) => console.error(url, err));
-    },
+var wrap = (url) => () => {
```
Comment (luqmaan) [inline client/js/utils/api.js]: Is there a more explicit name we can use instead of `wrap`?

**Comment by luqmaan** [comment]
:deciduous_tree: Great idea!

File: client/js/utils/api.js (hunk 1 above)
Comment (prestonp) [inline client/js/utils/api.js]: haha yeah that is a terrible name, let me rename

**Comment by aaronmgdr** [comment]
this seems great :deciduous_tree:

**Comment by prestonp** [comment]
I'm pretty bad at renaming, does that pattern make sense?

**Comment by luqmaan** [comment]
Seems good enough

**Comment by scascketta** [comment]
:+1: Good stuff.

**Pull request merged**